lib/runner.js

```diff
@@ -657,6 +661,30 @@ Runner.prototype.abort = function(){
 };
 
 /**
+ * Filter suites based on `isOnly` logic.
+ *
+ * @param {Array} suite
+ * @returns {Array}
+ * @api private
+ */
+function filterOnly(suite) {
+  // If it has `only` tests
+  if(suite.hasOnly) suite.tests = suite.hasOnly;
+  // If it doesn't has a suites make it running
+  // only if it mark as an `only`
+  if(!suite.suites.length) return suite.isOnly;
```

Comment:
Okay, as expected, it does not work with the following test file (or is this not a valid test file?):
```javascript
it('should not be executed', function() { })
it.only('is the only test that should be executed', function() { })
it('shouldn\'t be executed either', function() { })
```